File: save your girl.py
import random as r;
print("Welcome to : 'Save your girl'.")
print("coded by Rutheford")
import random as r
name=str(input("Write your name: ")).upper()
print(f"Hello {name}, Welcome to : 'Save your love'.")
print("Coded by Rutheford")
k= r.uniform(0, 12) #The random selection of the coodinate X
p= r.uniform(0, 12) #The random selection of the coodinate Y
)print("Your girlfriend/boyfriend was kidnapped by an zombie. You have to kill the zombie to save her/him. You only has 3 attempts")
print("The zombie's position is measured in a cartesian graphic")
if name != 'BIANCA':
 for g in range(1,4):
  x= float(input("x coordinate: ")) #The attempt of hitting the coodinate X
  y= float(input("y coordinate: ")) #The attempt of hitting the coodinate Y
  if((x-k)+(y-p))==0.01: #Limit of the difference between x and k (and respectivly) y and p, with that value gettin' near 0,01
    print(f"Congrats you nailed it, on the {g} attempt. Now she/he is safe. The coordinates were {k}, {p}.")
    break
  else:
   t=100*(x/k)
   v=100*(y/p)
   def percentage(t,v):
    if t>100:
     t=t-100
     return t
    if v>100:
     v=v-100
     return v
   print(f"You missed this time, try again. The precision is {t}% on the x coordinate and {v}% on the y coordinate. If the percentage is higher than 100, it means that the value is above the zombie's position.")
 if ((x-k)+(y-p))!=0.01 and g==3:
  print(f"Game over,{name}, you failed. The coordinates were {k}, {p}. ")
else: #the secret mode, that runs the same, but with more lifes
 for g in range(1,7):
   x= float(input("x coordinate: "))
   y= float(input("y coordinate: "))
   if((x-k)+(y-p))==0.01:
    print(f"Congrats, {name}, nailed it, on the {g} attempt. Now she/he is safe. The coordinates were {k}, {p}")
    break
   else:
    t=100*(x/k)
    v=100*(y/p)
    def percentage(t,v):
      if t>100:
       t=t-100
       return t
      if v>100:
       v=v-100
       return v
    print(f"You missed this time, The precision is {t}% on the x coordinate and {v}% on the y coordinate. If the percentage is higher than 100, it means that the value is above the zombie's position.")
 if ((x-k)+(y-p))!=0.01 and g==7:
    print(f"Game over, {name}, you failed. The coordinates were {k}, {p}.")

File: SaveYourGirl.cpp
#include <iostream>
#include <random>
using namespace std;
int main(){
 random_device rd;
 mt19937 gen(rd());
 uniform_real_distribution<float> distrib(0.0, 12.0);
 float coor1 = distrib(gen), coor2= distrib(gen), usimp1,usimp2;
 string name;
 int end;
 cout <<"Please, your name? ";
 cin >> name;
 cout<<"Welcome, "<< name <<", To : Save your girl/boy'."<<endl;
 cout<<"Your girlfriend/boyfriend was kidnapped by an zombie. You must rescue her/him, killing the zombie. you've only 3 shots."<<endl<<"Zombie's position equals to an point in the cartesian plan. (DECIMALS included)."<<endl;
   if (name!="Bianca"){
      end=3;}
   else{
      end=7;}
 for(int tries=1; tries<=(end);tries++)
   {cout<<"Input coordinate x: "; 
     cin>>usimp1;
     cout<<"Input coordinate y: "; 
     cin>>usimp2;
     if (((coor1 - usimp1) + (coor2 - usimp2))<=0.01) {
	    cout<<"Congrats,"<<name<<" ! Bull's eye on "<<tries<<"º shot/! Now, she/he's out of danger.";
	    return 0;}
     else{
      cout<<"It didn't, sorry. Precision of "<<(100*(usimp1/coor1))<<" % on x coordinate and "<<(100*(usimp2/coor2))<<" % on y coordinate. Try again."<<endl;}
      cout<<" If precision is bigger than 100%, its the excess"<<endl;
     if (((coor1-usimp1) + (coor2-usimp2))>0.01 && (tries==end)){
      cout<<"Well... Quite unfortunate, it's all over... \n The x value was: "<< coor1<<"\n And the y coord was: "<<coor2<<endl;}}
  return 0;}
